### PR TITLE
Add DMA support for ESP32-C2 and ESP32-S3

### PR DESCRIPTION
**Pull request by jessebraham**
ESP32-C2 was pretty simple, mostly just `#[cfg()]` gates.

The ESP32-S3 was a little more problematic, and I've made a bit of a mess getting things to work. Not sure how best to address these differences at this point.

### CODE REVIEW

File: esp-hal-common/src/dma/mod.rs [hunk 1]
```diff
@@ -20,7 +20,7 @@ pub enum DmaError {
 }
 
 /// DMA Priorities
-#[cfg(any(esp32c3, esp32s3))]
+#[cfg(any(esp32c2, esp32c3, esp32s3))]
 #[derive(Clone, Copy)]
 pub enum DmaPriority {
```
Comment (jessebraham) [inline esp-hal-common/src/dma/mod.rs:25]: Curious where these values came from?

In the ESP32-C2's ESP32-C3's TRMs it says:

> To ensure timely response to peripherals running at a high speed with low latency (such as SPI), the GDMA controller implements a fixed-priority channel arbitration scheme. That is to say, each channel can be assigned a priority from 0 ~ 9.

Comment (bjoernQ) [reply esp-hal-common/src/dma/mod.rs:25]: Oh yes ... I have no idea what I thought here. Mind changing it in your PR?

**Comment by bjoernQ** [comment]
Really sad ESP32-S3 introduced so much special handling - but great you implemented all this.
Code looks fine - just need to test and check it

File: esp-hal-common/src/dma/gdma.rs [hunk 2]
```diff
@@ -219,9 +361,11 @@ macro_rules! ImplChannel {
...
     use crate::dma::{private::*, *};
 
-    ImplChannel!(0);
-    ImplChannel!(1);
-    ImplChannel!(2);
+    impl_channel!(0);
+    #[cfg(not(esp32c2))]
+    impl_channel!(1);
+    #[cfg(not(esp32c2))]
+    impl_channel!(2);
```
Comment (bjoernQ) [inline esp-hal-common/src/dma/gdma.rs:368]: for ESP32-S3 we can add two more channels

**Comment by bjoernQ** [comment]
seems like on ESP32-C2 the SPI clock is just 50kHz not 100kHz - not something for this PR - we should create a new issue for that

**Comment by bjoernQ** [comment]
Great job! Would be nice if you could fix my mistake regarding the GDMA priorities and add two more DMA channels for S3

File: esp-hal-common/src/dma/mod.rs (hunk 1 above)
Comment (jessebraham) [reply esp-hal-common/src/dma/mod.rs:25]: Not at all, just wanted to make sure I wasn't missing something first haha

File: esp-hal-common/src/dma/gdma.rs (hunk 2 above)
Comment (jessebraham) [reply esp-hal-common/src/dma/gdma.rs:368]: Good catch, thanks! I'll add those in when I fix the priority.

**Comment by jessebraham** [comment]
@bjoernQ Thanks for the review, should be good now I think!

**Review by bjoernQ — APPROVED** [review]
LGTM - thanks for taking care of this

**Pull request merged**